### ISSUE TITLE
not fully detachAll

### ISSUE THREAD

**Issue by ScoobyMax** [orientechnologies/orientdb#8256]
## Expected behavior  

> OObjectEntitySerializer.detachAll(o, db, returnNonProxiedInstance, new HashMap<Object, Object>(), new HashMap<Object, Object>());

with returnNonProxiedInstance= true return full detached object

## Actual behavior  
return inner map as OTrackedMap

## Steps to reproduce  
save simple object:

```
public class SimpleObject{

	@Id
	private Object rid;

	@Version
	private Object version;

	private String objectId;

	private Map<String, String> templatePartsIds;

	public String getObjectId() {
		return objectId;
	}

	public void setObjectId(String objectId) {
		this.objectId = objectId;
	}

	public Map<String, String> getTemplatePartsIds() {
		return templatePartsIds;
	}

	public void setTemplatePartsIds(Map<String, String> templatePartsIds) {
		this.templatePartsIds = templatePartsIds;
	}
}
```

read it from DB and try to deatachall

**Comment by ScoobyMax** [comment]
Also with OTrackedList

**Comment by luigidellaquila** [comment]
Hi @ScoobyMax 

I just pushed a fix for this in 2.2.x branch, now I'm porting it to develop (3.0)
The fix will be released with 2.2.36 and 3.0.1

Thanks

Luigi
